src/librustc/mir/mod.rs

```diff
@@ -150,6 +150,12 @@ pub struct Mir<'tcx> {
     /// This is used for the "rust-call" ABI.
     pub spread_arg: Option<Local>,
 
+    /// Mark this MIR of a const context other than const functions as having converted a `&&` or
+    /// `||` expression into `&` or `|` respectively. This is problematic because if we ever stop
+    /// this conversion from happening and use short circuiting, we will cause the following code
+    /// to change the value of `x`: `let mut x = 42; false && { x = 55; true };`
+    pub const_can_have_let_mut_bindings: bool,
```

Comment:
This is not what I had in mind - that is, not `let mut` specifically, I just want to disable all new features in constants where `&&` and `||` were *lowered* to `&` and `|`.
And I wouldn't want the flag to say what it does, but *what had transpired*.

So to me, the poisoning of the MIR is about "did we destroy control-flow information?", which means we can't assume linear controlflow is an accurate representation of what the user wrote.